### PR TITLE
BUG: correct the scaling in the floating-point slop test.

### LINKED ISSUE #11587: Missing filled contours when using contourf
I am trying to plot plot a function over a 2D domain using `contourf`. Unfortunately, my first attempt did not work out very well. There was a region in the plot that was unexpectedly not covered by any contours. For debugging purposes, I have reduced the problem to the smallest dataset that I could find that also reveals the issue with missing filled contours:

```
import matplotlib.pyplot as plt
import numpy as np

v = np.array([0, 1, 2, 3])
x, y = np.meshgrid(v, v)

z = np.array([[5.5e-14, 5.5e-14, 5.5e-14, 5.5e-14],
              [2e-13, 2e-13, 2e-13, 2e-13],
              [2.2e-13, 2.2e-13, 2.2e-13, 2.2e-13],
              [0, 0,0, 0]])
fig, ax = plt.subplots()
cntr = ax.contourf(x, y, z)
fig.colorbar(cntr, ax=ax)
plt.show()
```
figure_1

as you can see there are missing contours from `y = 1.5` to approximately `y = 2.0`.

Another strange thing I observed: If I normalize the `z` matrix by multiplying by e.g. `1e14` before plotting it works fine.

**Matplotlib version**

  * Operating system: Ubuntu 18.04
  * Matplotlib version: 2.2.2 (Installed with `pip`)
  * Matplotlib backend: TkAgg
  * Python version: 3.6.1

Note: This question was first asked on stackoverflow.com: [Missing filled contours when using contourf](https://stackoverflow.com/q/51210497/2173773)

### PR DESCRIPTION
**Pull request by efiring**
Closes #11587.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
Ooops, though this fails `test_axes.py::test_autoscale_tiny_range`

**Comment by efiring** [comment]
That blows the lid off the worm can.  Working on it.

**Comment by efiring** [comment]
I think this is getting close to a reasonable solution.  `Base` is confusing and broken (and we use only part of it), so I would like to deprecate it in favor of the private `_Edge_integer` class, thereby reducing the public API; this is really implementation detail that we should be free to change.  `_Edge_integer` probably can be improved (and maybe someone has a better name), but it is at least doing the right general thing in adapting its `closeto` to the ratio between the offset and the step, which `Base` does not do.

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -1766,6 +1765,37 @@ def scale_range(vmin, vmax, n=1, threshold=100):
     return scale, offset
 
 
+class _Edge_integer:
```
Comment (jklymak) [inline lib/matplotlib/ticker.py]: Needs some documentation!

File: lib/matplotlib/ticker.py [hunk 2]
```diff
@@ -1918,13 +1948,14 @@ def _raw_ticks(self, vmin, vmax):
                 step = max(1, step)
             best_vmin = (_vmin // step) * step
 
-            low = np.round(Base(step).le(_vmin - best_vmin) / step)
-            high = np.round(Base(step).ge(_vmax - best_vmin) / step)
-            ticks = np.arange(low, high + 1) * step + best_vmin + offset
-            nticks = ((ticks <= vmax) & (ticks >= vmin)).sum()
+            edge = _Edge_integer(step, offset)
```
Comment (jklymak) [inline lib/matplotlib/ticker.py]: Similarly, this whole block could use a couple lines of comment.  I looked at this the other day, and worked my way through it, but on its face it looks more complicated than whats needed.

**Comment by efiring** [comment]
I think this is ready to go now, with the possible exception of a next_api_changes note for the deprecation of `ticker.Base` and `ticker.closeto`.

**Comment by efiring** [comment]
I would be happy to move the tags up to 3.0 if you are interested in squeezing this into 3.0.  Leaving it as 3.1 is OK, too.

**Pull request merged**